### DISCUSSION TITLE
Select the quantity of comented lines at the end of a G-code

### DISCUSSION THREAD

**Discussion by jfestrada** [supermerill/SuperSlicer#1448] [Community contribution]
Currently the firmware of the MK3S is taking care of the last 20KB of the file in order to search the M84 just to check the integrity of the file and be sure that the file is complete. Whe you are using multimaterial the quantity of comented lines is too high in  SuperSlicer (arount 24KB) so the printer will allways alert to you that the file is incomplete. 
So there are many ways to solve this situation, obviously the "easy way" is to increase the number of bytes to be checked by the printer, this option force to you to recompile the firmware. The other way could be create an option to limit the number of comented lines that will be added at the end of the G-Code, for example using a dropdownlist with the typical options of simple, expert, advanced.
Would be nice have the opportunity to set this option from the machine profile.  What do you think? Has anyone found this situation previously?

**Comment by supermerill** [comment]
Will do something with the gcode writer rewrite.

For now, should you be good by adding it to the start_gcode (as a comment) ?

**Reply by jfestrada** [reply]
Not really, the firmware checks the last 20000 bytes of the file. 

![imagen](https://user-images.githubusercontent.com/7290345/127849923-b5613594-b367-485f-b7c4-7c8e5375487b.png)

Anyway if you add the M84 as a comment will not be take into account by the cmdqueue so still a not valid solution.

**Reply by supermerill** [reply]
Understand, it doesn't check the comments but what's before it.

Just a note: you should have opened an issue from the get-go.

Not possible to circumvent it right now. Only two way:
 * create a setting and a kill switch
 *  using a post-processing script that remove the comments at the end of the file.

**Reply by jfestrada** [reply]
Ok, I will open the issue, by now for VERTEX printers we are testing the changes on the firmware in order to read the last 27000 bytes of the file so we will be covered, but the PRUSA MK3S will still have the issue, maybe the kill switch will be the best option.
